R/downloadableReactTable.R

```diff
@@ -144,37 +142,72 @@ downloadableReactTableUI <- function(id,
                      output$reactTableOutputID <- reactable::renderReactable({ NULL })
                  } else {
-                     table_react_params <- shiny::reactiveValues(table_data = NULL)
+                     table_react_params <- shiny::reactiveValues(table_data        = NULL,
+                                                                 pre_selected_rows = NULL)
+                     if (!is.null(pre_selected_rows) && !is.function(pre_selected_rows)) {
+                         message("'pre_selected_rows' parameter must be a function or reactive expression. Setting default value NULL.")
+                         pre_selected_rows <- NULL
+                     }
+
                      shiny::observe({
                          if (!is.data.frame(table_data())) {
                              table_data <- shiny::reactiveVal(data.frame(table_data()))
                          }
                          table_react_params$table_data <- table_data()
                      })
+
+                     shiny::observe({
+                         table_react_params$pre_selected_rows <- NULL
+                         if ((is.null(selection_mode) || !(tolower(selection_mode) %in% c("single", "multiple"))) &&
+                              !is.null(pre_selected_rows)) {
+                             message("'selection_mode' parameter must be either 'single'or 'multiple' to use 'pre_selected_rows' param. Setting default value NULL.")
+                         } else if (!is.null(pre_selected_rows) && !is.numeric(pre_selected_rows())) {
+                             message("'pre_selected_rows' parameter must function or reactive expression must return numeric vector. Setting default value NULL.")
+                         } else if (!is.null(pre_selected_rows)) {
+                             table_react_params$pre_selected_rows <- pre_selected_rows()
+                         }
+
+                     })
                      output$reactTableOutputID <- reactable::renderReactable({
                          table_output <- NULL
+
                          if (!all(is.na(table_react_params$table_data)) &&
                              !is.null(table_react_params$table_data) &&
                              (NCOL(table_react_params$table_data) > 0)) {
-                             table_output <- reactable::reactable(data = table_react_params$table_data)
+                             row_selection_mode <- NULL
+                             if (!is.null(selection_mode) && (tolower(selection_mode) %in% c("single", "multiple"))) {
+                                 row_selection_mode <- tolower(selection_mode)
+
+                                 if ((row_selection_mode == "single") && (length(table_react_params$pre_selected_rows) > 1)) {
+                                     message("'selection_mode' is 'single' only first value of 'pre_selected_rows' will be used")
```

Comment:
"when 'selection_mode' is 'single' only first value of 'pre_selected_rows' will be used"

Reply:
done